ecoli/processes/metabolism_redux.py

```diff
@@ -0,0 +1,499 @@
+                    # TODO (Matt -> Cyrus): These should not be using a divider. Mass listener should run before metabolism after division.
+                    'cell_mass': {'_default': 0.0,
+                                  '_divider': 'split'},
+                    'dry_mass': {'_default': 0.0,
+                                 '_divider': 'split'}},
+
+                'fba_results': {
+                    'estimated_fluxes': {'_default': {}, '_updater': 'set', '_emit': True},
+                    'estimated_homeostatic_dmdt': {'_default': {}, '_updater': 'set', '_emit': True},
+                    'target_homeostatic_dmdt': {'_default': {}, '_updater': 'set', '_emit': True},
+                    'target_kinetic_fluxes': {'_default': {}, '_updater': 'set', '_emit': True},
+                    'target_kinetic_bounds': {'_default': {}, '_updater': 'set', '_emit': True},
+                    'estimated_exchange_dmdt': {'_default': {}, '_updater': 'set', '_emit': True},
+                    'estimated_intermediate_dmdt': {'_default': {}, '_updater': 'set', '_emit': True},
+                    'maintenance_target': {'_default': {}, '_updater': 'set', '_emit': True},
+                    'solution_fluxes': {'_default': {}, '_updater': 'set', '_emit': True},
+                    'solution_dmdt': {'_default': {}, '_updater': 'set', '_emit': True},
+                    'solution_residuals': {'_default': {}, '_updater': 'set', '_emit': True},
+                    'time_per_step': {'_default': 0.0, '_updater': 'set', '_emit': True},
+                },
+            },
+            'first_update': {
+                '_default': True,
+                '_updater': 'set',
+                '_divider': {'divider': 'set_value', 'config': {'value': True}},
+            },
+            'evolvers_ran': {'_default': True},
+        }
+
+    def update_condition(self, timestep, states):
+        return states['evolvers_ran']
+
+    def next_update(self, timestep, states):
+        # Skip t=0
+        if states['first_update']:
+            return {'first_update': False}
+
+        # metabolites not in either set are constrained to zero uptake.
+        UNCONSTRAINED_UPTAKE = states['environment']['exchange_data']['unconstrained']
+        CONSTRAINED_UPTAKE =  states['environment']['exchange_data']['constrained']
+        self.allowed_exchange_uptake = list(UNCONSTRAINED_UPTAKE) + list(CONSTRAINED_UPTAKE.keys())
+        self.disallowed_exchange_uptake = list(set(self.exchange_molecules) - set(self.allowed_exchange_uptake))
+        self.exchange_molecules = list(set(self.exchange_molecules).union(set(self.allowed_exchange_uptake)))
+
+        # set up network flow model exchanges and uptakes
+        self.network_flow_model.set_up_exchanges(self.exchange_molecules, self.allowed_exchange_uptake)
+
+        # extract the states from the ports
+        current_metabolite_counts = states['metabolites']
+        self.timestep = self.calculate_timestep(states)
```

Comment:
The step needs to know this since it uses it to scale all rates to per second rates. 

That seems like a good idea - I guess for now there's no need to solve that until partitioning is removed, or we introduce a process with smaller timestep than those processes.

Reply:
Added warning in the process.